lib/src/builder/README.md

```diff
@@ -52,26 +45,74 @@ own components by viewing the generated .over_react.g.dart file.
     
-2. The builder creates the consumable implementation of the props class. This is the class which contains concrete 
-getters/setters (via the `_$FooPropsAccessorsMixin` class)
+    In this case:
+    1. The builder identifies the compilation unit members `Foo`, `FooProps`, `FooComponent`, which all look like pieces of over_react boilerplate.
+    2. The builder sees that `FooProps` meets the requirements for declaring a props mixin, and thus  can confidently proceed with code generation for it.
+        - Props mixins need code generation regardless of whether they're associated with a component, since they can be declared on their own.
+    3. The builder starts with the factory, and looks for other members with matching names that it could use to create a complete group that forms a class component. It finds all the pieces it needs:
+        - [x] a factory: ***Foo***
+        - [x] either a props mixin or a concrete props class:  ***Foo***Props
+        - [ ] either a state mixin or a concrete state class (optional)
+        - [x] a component class: ***Foo***Component
+    2. The builder checks to see if the group is really an over_react declaration that needs code generation. It sees that the factory is referencing a generated variable (`Foo = _$Foo`) and thus can confidently say that the factory needs code generation. Because of that, it proceeds with code generation for all the members in this group. 
+   
+   The builder has recognized our code as a class component that requires code generation! If it hadn't, the builder would have gone through similar logic to identify other types of declarations like function components and props map views, and perform similar code generation to wire them up.
+   
+   Next up, let's see what code generation looks like for our class component.
+    
+2. The builder generates a version of the props mixin which contains concrete 
+getters/setters. 
     ```dart
-   class FooProps extends _$FooProps with _$FooPropsAccessorsMixin {
-     static const PropsMeta meta = _$metaForFooProps;
+   mixin $FooProps on FooProps {
+     int get foo => props['FooProps.foo'];
+     set foo(int value) => props['FooProps.foo'] = value;
    } 
     ```
+       
+2. The builder creates a concrete props implementation class, which mixes in the generated props mixin:
+    ```dart
+    class _$$FooProps extends UiProps with FooProps, $FooProps {
+      ...   
+    }
+    ```
+    This concrete class is what will be used everywhere for this component's typed props (returned from the `UiFactory`, available via `props` inside `render`).
+   
+    Note that in the above example, our component has props declared via just a props mixin, which is a shorthand syntax. For components with concrete props classes, the implementation is similar, and mixins in all generated mixins corresponding to the props mixins used:
```

Comment:
I think this is clearer. Please tweak if it's not 100% accurate tho.
```suggestion
    Note that in the above example, our component has props declared via just a props mixin, which is a shorthand syntax for when no additional props are needed from other mixins. For components with concrete props classes which mix in other props, the implementation is similar. The generated concrete props class mixes in all generated mixins corresponding to the props mixins used:
```

Reply:
Committed!